packages/web-app-files/src/composables/scrollTo/useScrollTo.ts

```diff
@@ -21,30 +21,16 @@ export const useScrollTo = (): ScrollToResult => {
     return queryItemAsString(unref(detailsQuery))
   })
 
-  const scrollToResource = (resource) => {
+  const scrollToResource = (resourceId: Resource['id']) => {
     const resourceElement = document.querySelectorAll(
-      `[data-item-id='${resource.id}']`
+      `[data-item-id='${resourceId}']`
     )[0] as HTMLElement
 
     if (!resourceElement) {
       return
     }
 
-    // bottom reached
-    if (resourceElement.getBoundingClientRect().bottom > window.innerHeight) {
-      resourceElement.scrollIntoView(false)
-      return
-    }
-
-    const topbarElement = document.getElementsByClassName('files-topbar')[0] as HTMLElement
-    // topbar height + th height + height of one row = offset needed when scrolling top
-    const topOffset = topbarElement.offsetHeight + resourceElement.offsetHeight * 2
-
-    // top reached
-    if (resourceElement.getBoundingClientRect().top < topOffset) {
-      const fileListWrapperElement = document.getElementsByClassName('files-view-wrapper')[0]
-      fileListWrapperElement.scrollBy(0, -resourceElement.offsetHeight)
-    }
+    resourceElement.scrollIntoView({ behavior: 'smooth', block: 'center' })
```

Comment:
Not sure why all the other code was necessary tbh - scrolling to a resource and centering it on screen seems pretty straightforward, and if we need a "scroll to top" we should implement it explicitly?

Reply:
doesn't work with active pagination though. if you're on page 3 and a just created folder is sorted into page 17 you would need to navigate to that page before trying to scroll...

Reply:
Ah damn, I knew there was something I was missing (because otherwise that issue would've been resolved long ago already I suppose)

Reply:
fine to split it into two issues btw. for this PR: assume that using `scrollToResource` is sufficient and declare it not checking pagination as bug. Then solve using pagination in `scrollToResource` as a separate PR.

Reply:
> Not sure why all the other code was necessary tbh - scrolling to a resource and centering it on screen seems pretty straightforward, and if we need a "scroll to top" we should implement it explicitly?

It was there for the keyboard navigation. Now when navigating via keyboard, it always immediately scrolls to the next resource. Before, it only scrolled when the resource was outside of the viewport. Both works, but I actually like the latter because it does not feel so "floaty".